### PR TITLE
Add Blade, PDU, Tor actual views to the inventory client

### PR DESCRIPTION
**Pull request by Jim3Things**

### CODE REVIEW

File: simulation/pkg/protos/inventory/actual.Helpers.go [hunk 1]
```diff
@@ -29,9 +33,60 @@ func (x *Actual_ActiveTimer) Clone() *Actual_ActiveTimer {
...
+	for k, c := range cables {
+		res[k] = c.Clone()
+	}
+
+	return res
+}
+
+func EqualActualCables(x map[int64]*Actual_Cable, y map[int64]*Actual_Cable) bool {
+	if x == nil || y == nil {
+		return x == nil && y == nil
```
Comment (MikeCJohnsonUk) [inline simulation/pkg/protos/inventory/actual.Helpers.go:78]: Any reaon(s) for not using the "return x == y" form?

File: simulation/internal/clients/inventory/actual_test.go [hunk 2]
```diff
@@ -56,13 +181,21 @@ func (*actualTestSuite) pduID(ID int64) int64            { return ID }
 func (*actualTestSuite) torID(ID int64) int64            { return ID }
 func (*actualTestSuite) bladeID(ID int64) int64          { return ID }
 
+func (ts *actualTestSuite) SpecializeName(name string) string {
+	return strings.Replace(name, "$TEST$", ts.TestName(), -1)
```
Comment (MikeCJohnsonUk) [inline simulation/internal/clients/inventory/actual_test.go:185]: Possible use of ReplaceAll(x, y, z) over Replace(x, y, z, -1)

File: simulation/internal/clients/inventory/blade.go [hunk 3]
```diff
@@ -124,12 +120,7 @@ func (b *Blade) mapCommonErrors(keys keys, err error) error {
 // to the view-specific data fields.
 //
 type BladeViewDefinition struct {
-	revisionInfo
-	itemStoreLeaf
-
-	keys
-
-	blade *Blade
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply simulation/internal/clients/inventory/blade.go:132]: That's right.  These are not subsumed into commonLeafView.

File: simulation/pkg/protos/inventory/actual.Helpers.go (hunk 1 above)
Comment (Jim3Things) [reply simulation/pkg/protos/inventory/actual.Helpers.go:78]: Go didn't like that for maps, so I had to do it the hard way...

File: simulation/internal/clients/inventory/actual_test.go (hunk 2 above)
Comment (Jim3Things) [reply simulation/internal/clients/inventory/actual_test.go:185]: Cool.  Mind if I take that on when I do racks?

File: simulation/pkg/protos/inventory/actual.Helpers.go (hunk 1 above)
Comment (MikeCJohnsonUk) [reply simulation/pkg/protos/inventory/actual.Helpers.go:78]: I figured that might be it.

Ok.

File: simulation/internal/clients/inventory/actual_test.go (hunk 2 above)
Comment (MikeCJohnsonUk) [reply simulation/internal/clients/inventory/actual_test.go:185]: Makes sense.

Ok.

File: simulation/internal/clients/inventory/blade.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply simulation/internal/clients/inventory/blade.go:132]: Ok.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**